discord/invite.py

```diff
@@ -372,6 +377,10 @@ def url(self):
         """:class:`str`: A property that retrieves the invite URL."""
         return self.BASE + '/' + self.code
 
+    def is_stream_invite(self):
+        """:class:`bool`: Whether this invite is to a stream."""
+        return self._is_stream_invite
```

Comment:
I'm not sure what interface is worth exploring. I'll walk through my thought process and maybe come to a conclusion near the end.

There are two designs that can be expressed by the user:

```py
invite.target_user_type is InviteTargetUser.stream
```

or, the easier to read implementation given here:

```py
invite.is_stream_invite()
```

The first design is more explicit towards how the Discord API functions but the second one is admittedly a lot more user friendly. The second design could replace the first design entirely and we could completely forego the Enum approach, storing an integer directly within the model. This leads to better performance but it means that the API for checking the type would be entirely located within the `Invite` model. I think for both documentation reasons and ease of implementability, the second design with this interface is probably better.

Reply:
Thanks for the feedback, the possible expansion of the target_user_type sounds like something that is worth planning for now, the Enum route you suggest will likely be more robust. In the future if discord expands the `target_user_type`, we don't really want a helper method like this for every type of hypothetical invite type, especially since nothing similar comes to mind for other Discord.py models. In short, I think removing the is_stream_invite method and letting end users compare the enum themselves would be better